src/pivot.coffee

```diff
@@ -780,26 +786,26 @@ callWithJQuery ($) ->
                 tr.appendChild td
 
             #create rightmost row totals cell/s
-            createTotalsCell = (totalAggregator) ->
+            createTotalsCell = (totalAggregator, aggIdx) ->
                 val = totalAggregator.value()
                 td = document.createElement("td")
                 td.className = "pvtTotal rowTotal"
                 td.textContent = totalAggregator.format(val)
                 if heatmappers?
                     td.style.backgroundColor = heatmappers.rowTotals(val)
                 if getClickHandler?
-                    td.onclick = getClickHandler(val, rowKey, [])
+                    td.onclick = getClickHandler(val, rowKey, [], aggIdx)
```

Comment:
That's a good question. The `_metric` key is normally present in the 3rd argument (colKey), as it is added to the colAttrs automatically. Since for the row totals, we aren't passing in any colAttrs, we have to pass the aggregator index instead as a substitute. This makes me think that a potential option would be to create a colKey and pass that into this function instead. That would look something like this:
```
colKey = [{@multiAggAttr: aggIdx}]
td.onclick = getClickHandler(val, rowKey, colKey)
```
What do you think?

Reply:
_metrics can be moved into the row attributes just like any other attribute. 

You can see the bug if you do that, and click on a column total. The aggregator index won't be passed into the drill down.

It seems like you're hesitant to update the column totals click handler as I suggested. How come?